genie-web/src/main/java/com/netflix/genie/web/tasks/job/JobCompletionService.java

```diff
@@ -268,29 +268,31 @@ private Void updateJob(final Job job, final JobFinishedEvent event, final Map<St
 
     /**
      * An external fail-safe mechanism to clean up processes left behind by the run.sh after the
-     * job is killed or failed.
+     * job is killed or failed. This method is a no-op for jobs whose status is INVALID.
      *
      * @param jobId The id of the job to cleanup processes for.
      */
     private void cleanupProcesses(final String jobId) {
         try {
-            this.jobSearchService.getJobExecution(jobId).getProcessId().ifPresent(pid -> {
-                try {
-                    final CommandLine commandLine = new CommandLine(JobConstants.UNIX_PKILL_COMMAND);
-                    commandLine.addArgument(JobConstants.getKillFlag());
-                    commandLine.addArgument(Integer.toString(pid));
-                    executor.execute(commandLine);
-
-                    // The process group should not exist and the above code should always throw and exception.
-                    // If it does not then the bash script is not cleaning up stuff well during kills
-                    // or the script is done but child processes are still remaining. This metric tracks all that.
-                    processGroupCleanupFailureRate.increment();
-                } catch (final Exception e) {
-                    log.debug("Received expected exception. Ignoring.");
-                }
-            });
+            if (!this.jobSearchService.getJobStatus(jobId).equals(JobStatus.INVALID)) {
```

Comment:
Have we gotten the job status anywhere else that we can pass in and save from making yet another DB call here?